### PR TITLE
Closes #259

### LINKED ISSUE #259: Add "Delete All" and "Delete Selected" menu to items where multiple child items can be added.
For items where multiple child items can be added, add menu "Delete All" and "Delete Selected", so that user can delete items easily when there are many.

The items to add the menus above are:

* "Boundary Condition Settings" in pre-processing window
* "Boundary Condition" in pre-processing window
* "Geographic data" in pre-processing window
* "Measured data" in pre-processing window and post2d window
* "Distance measure" in pre-processing window and post 2d window
* "Background Images" in pre-processing window and post2d window

### PR DESCRIPTION
**Pull request by kskinoue0612**
You can test with the file below in issue-259.zip:

## boundary_condition.ipro

* "Boundary Condition Settings" in pre-processing window
* "Boundary Condition" in pre-processing window

## post.ipro

* "Geographic data" in pre-processing window
* "Measured data" in pre-processing window and post2d window
* "Distance measure" in pre-processing window and post 2d window
* "Background Images" in pre-processing window and post2d window

[issue-259.zip](https://github.com/i-RIC/prepost-gui/files/5136402/issue-259.zip)

### CODE REVIEW

File: libs/pre/datamodel/preprocessorbcdataitem.h [hunk 1]
```diff
@@ -82,9 +87,7 @@ public slots:
 	bool showDialog();
 
 private slots:
-	void setModified(bool modified = true) override {
-		PreProcessorDataItem::setModified(modified);
-	}
+	void setModified(bool modified = true);
```
Comment (scharlton2) [inline libs/pre/datamodel/preprocessorbcdataitem.h:90]: Hi Keisuke,

Shouldn't this still have override attached to it?

Thanks,
Scott

**Comment by kskinoue0612** [comment]
Hi Scott,

Thanks for the comment. You are right, I've removed "override" by mistake.
I've pushed a commit to fix it.

Thanks,
Keisuke
